src/fluree/db/query/json_ld/response.cljc

```diff
@@ -75,31 +75,44 @@
             acc))))))
 
 (defn crawl-ref-item
-  [db compact-fn ref-flake sub-select cache fuel-vol max-fuel depth-i]
+  [db compact-fn flake-sid sub-select cache fuel-vol max-fuel depth-i]
   (go-try
-    (let [sub-flakes (<? (query-range/index-range db :spot = [(flake/o ref-flake)]))]
+    (let [sub-flakes (<? (query-range/index-range db :spot = [flake-sid]))]
       (<? (flakes->res db cache compact-fn fuel-vol max-fuel sub-select depth-i sub-flakes)))))
 
-(defn crawl-ref
-  "A sub-selection (graph crawl) exists, generate results."
-  [db compact-fn p-flakes sub-select cache fuel-vol max-fuel depth-i]
+
+(defn add-reverse-specs
+  "When @reverse variables are present, crawl for the reverse specs."
+  [db cache compact-fn fuel-vol max-fuel {:keys [reverse] :as select-spec} depth-i flakes]
   (go-try
-    (loop [[next-flake & r] p-flakes
-           acc []]
-      (if next-flake
-        (let [sub-flakes (<? (query-range/index-range db :spot = [(flake/o next-flake)]))
-              res        (<? (flakes->res db cache compact-fn fuel-vol max-fuel sub-select depth-i sub-flakes))]
-          (recur r (conj acc res)))
-        (if (= 1 (count acc))
-          (first acc)
+    (let [sid (flake/s (first flakes))]
+      (loop [[reverse-item & r] (vals reverse)
+             acc {}]
+        (if reverse-item
+          (let [{:keys [id as spec]} reverse-item
+                sub-flakes (<? (query-range/index-range db :opst = [sid id]))
+                result     (loop [[ref-sid & r] (map flake/s sub-flakes)
+                                  acc-item []]
+                             (if ref-sid
+                               (let [result (if spec
+                                              ;; have a sub-selection
+                                              (<? (crawl-ref-item db compact-fn ref-sid spec cache fuel-vol max-fuel (inc depth-i)))
+                                              ;; no sub-selection, just return IRI
+                                              (or (get @cache ref-sid)
+                                                  (<? (cache-sid->iri db cache compact-fn ref-sid))))]
+                                 (recur r (conj acc-item result)))
+                               (if (= 1 (count acc-item))
+                                 (first acc-item)
+                                 acc-item)))]
+            (recur r (assoc acc as result)))
           acc)))))
 
 
 ;; TODO - check for @reverse
```

Comment:
Looks like we can remove this now.

Reply:
I don't think so, but might not know exactly what you mean.

Reply:
I thought this was a TODO to add reverse crawl support, which this commit does, but nvm.

Reply:
AHH, sorry didn't really the comment was for the comment, it looked like it was for the new code.

I'll update this in the other PR I have open.